### PR TITLE
DEV-2963

### PR DESCRIPTION
**Pull request by stefvanlieshout**

### CODE REVIEW

File: src/main/java/com/hartwig/snpcheck/PerlVcfComparison.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 public class PerlVcfComparison implements VcfComparison {
 
     @Override
-    public Result compare(final Run run, final Blob refVcf, final Blob valVcf) {
+    public Result compare(final Run run, final Blob refVcf, final Blob valVcf, final Boolean alwaysPass) {
```
Comment (nedleitch) [inline src/main/java/com/hartwig/snpcheck/PerlVcfComparison.java:16]: I prefer to stick with the primitive `boolean` wherever possible as it avoids the possibility of `null`, which avoids surprises. Not a big deal in this instance but maybe something to keep in mind.

File: src/main/java/com/hartwig/snpcheck/SnpCheckMain.java [hunk 2]
```diff
@@ -44,6 +44,11 @@ public class SnpCheckMain implements Callable<Integer> {
             description = "Mark all events as validated without actually validating against the snpcheck vcf.")
     private boolean passthru;
 
+    @CommandLine.Option(names = { "--allways_pass" },
+                        defaultValue = "false",
+                        description = "Run the actual snpcheck in allways pass mode.")
```
Comment (nedleitch) [inline src/main/java/com/hartwig/snpcheck/SnpCheckMain.java:49]: Nit: `always` not `allways`.

File: src/main/java/com/hartwig/snpcheck/VcfComparison.java [hunk 3]
```diff
@@ -2,13 +2,12 @@
...
-import com.hartwig.api.model.Sample;
 
 public interface VcfComparison {
     enum Result {
         PASS,
         FAIL
     }
 
-    Result compare(final Run run, final Blob refVcf, final Blob valVcf);
+    Result compare(final Run run, final Blob refVcf, final Blob valVcf, final Boolean alwaysPass);
```
Comment (nedleitch) [inline src/main/java/com/hartwig/snpcheck/VcfComparison.java:12]: Same comment on primitive vs object.

File: src/test/java/com/hartwig/snpcheck/PerlVcfComparisonExecutionTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+public class PerlVcfComparisonExecutionTest {
+    private static final String USER_DIR = System.getProperty("user.dir");
+    private static final String BAD_GENOTYPE_VCF = USER_DIR + "/src/test/resources/bad_genotype.vcf";
+    private static final String GOOD_GENOTYPE_VCF_1 = USER_DIR + "/src/test/resources/good1_genotype.vcf";
+    private static final String GOOD_GENOTYPE_VCF_2 = USER_DIR + "/src/test/resources/good2_genotype.vcf";
+
+    @Test
+    public void goodSnpcheckYieldsPass() {
+        Result result = new PerlVcfComparisonExecution().execute(GOOD_GENOTYPE_VCF_1, GOOD_GENOTYPE_VCF_2, Boolean.FALSE);
+        assertThat(result).isEqualTo(Result.PASS);
```
Comment (nedleitch) [inline src/test/java/com/hartwig/snpcheck/PerlVcfComparisonExecutionTest.java:16]: I'd rather see these inlined, no need for the variable.

**Review by pauldwolfe — APPROVED** [review]

**Pull request merged**